sdk/ml/azure-ai-ml/azure/ai/ml/_schema/assets/package/base_environment_source.py

```diff
@@ -0,0 +1,23 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=unused-argument,no-self-use
+
+import logging
+from marshmallow import fields, post_load
+from azure.ai.ml._schema.core.schema import PathAwareSchema
+
+
+module_logger = logging.getLogger(__name__)
+
+
+class BaseEnvironmentSourceSchema(PathAwareSchema):
+    type = fields.Str()
+    resource_id = fields.Str()
+
+    @post_load
+    def make(self, data, **kwargs):
+        from azure.ai.ml.entities import BaseEnvironmentId
```

Comment:
Should we name the schema as {entityName}Schema? Do we have a pattern we follow?

Reply:
Looking at the next couple files, it seems this should be called "BaseEnvironmentIdSchema"

Reply:
We don't have a strong naming convention here and it does not really matter much as these names are not exposed to public